src/chocolatey/infrastructure.app/services/PowershellService.cs

```diff
@@ -59,23 +60,31 @@ public PowershellService(IFileSystem fileSystem, CustomString customImports)
             _customImports = customImports;
         }
 
-        public void noop_action(PackageResult packageResult, CommandNameType command)
+        private string get_script_for_action(PackageResult packageResult, CommandNameType command)
         {
             var file = "chocolateyInstall.ps1";
             switch (command)
             {
                 case CommandNameType.uninstall:
                     file = "chocolateyUninstall.ps1";
                     break;
+
+                case CommandNameType.upgrade:
+                    file = "chocolateyBeforeModify.ps1";
+                    break;
             }
 
             var packageDirectory = packageResult.InstallLocation;
-            var installScript = _fileSystem.get_files(packageDirectory, file, SearchOption.AllDirectories).Where(p => !p.to_lower().contains("\\templates\\"));
-            if (installScript.Count() != 0)
-            {
-                var chocoInstall = installScript.FirstOrDefault();
+            var installScript = _fileSystem.get_files(packageDirectory, file, SearchOption.AllDirectories).FirstOrDefault();

```

Comment:
Hmmm... that's worrying. The removed code doesn't look at all familiar to me (I don't remember seeing anything related to the templates folder previously, even in any "merge conflict" scenario). Thanks for doing the fixup.